[CI] CcrSeqNoPruningIT testSeqNoPartiallyRetainedByCcrLease failing

**Build Scans:**
- [elasticsearch-intake #40692 / part2](https://gradle-enterprise.elastic.co/s/dkuglfkvaxzwo)
- [elasticsearch-pull-request #151324 / part-2](https://gradle-enterprise.elastic.co/s/wuwnr3l2agcp4)
- [elasticsearch-pull-request #151285 / part-2](https://gradle-enterprise.elastic.co/s/he4lbdni7xazg)
- [elasticsearch-pull-request #151258 / part-2](https://gradle-enterprise.elastic.co/s/f6gosc2nxntem)
- [elasticsearch-periodic-platform-support #12310 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/fuipdbcvkjfi4)
- [elasticsearch-periodic #12339 / openjdk21_checkpart2_java-fips-matrix](https://gradle-enterprise.elastic.co/s/qmmc3vk6ylh3w)
- [elasticsearch-pull-request #150877 / part-2](https://gradle-enterprise.elastic.co/s/xavcx3ddp5gyc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ccr:internalClusterTest" --tests "org.elasticsearch.xpack.ccr.CcrSeqNoPruningIT.testSeqNoPartiallyRetainedByCcrLease" -Dtests.seed=465B4CCE0142F975 -Dtests.locale=ckb -Dtests.timezone=Africa/Gaborone -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ccr.CcrSeqNoPruningIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSeqNoPartiallyRetainedByCcrLease'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: [xvmjmlmuq][0] _seq_no doc values should be empty
Expected: <2147483647>
     but: was <291>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testSeqNoPartiallyRetainedByCcrLease
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 7 failures in test testSeqNoPartiallyRetainedByCcrLease (1.3% fail rate in 560 executions)
- [main] 4 failures in step part-2 (1.3% fail rate in 310 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 307 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -374,9 +374,6 @@ tests:
   method: "builds distribution from branches via archives extractedAssemble [bwcDistVersion: 8.2.1, bwcProject: major3, expectedAssembleTaskName:
     extractedAssemble, #2]"
   issue: https://github.com/elastic/elasticsearch/issues/150410
-- class: org.elasticsearch.xpack.ccr.CcrSeqNoPruningIT
-  method: testSeqNoPartiallyRetainedByCcrLease
-  issue: https://github.com/elastic/elasticsearch/issues/150435
 - class: org.elasticsearch.xpack.esql.action.FileSourceSecretDecryptionAcrossNodesIT
   method: testJoinedDataNodeDecryptsTheSecret
   issue: https://github.com/elastic/elasticsearch/issues/150440

